axelrod/tests/strategies/test_titfortat.py

```diff
@@ -563,3 +563,52 @@ def test_strategy(self):
         actions = [(C, C), (C, C), (C, D), (C, D), (D, C), (D, D), (D, D),
                     (D, C), (D, C), (C, D), (C, D)]
         self.versus_test(opponent, expected_actions=actions)
+
+class TestAlexei(TestPlayer):
+    """
+    Tests for the Alexei strategy
+    """
+
+    name = "Alexei: ('D',)"
+    player = axelrod.Alexei
+    expected_classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': {'length'},
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        self.first_play_test(C)
+        self.second_play_test(rCC=C, rCD=D, rDC=C, rDD=D)
+
+        actions = [(C, C), (C, D), (D, C), (C, D), (D, C)]
+        self.versus_test(axelrod.Alternator(), expected_actions=actions)
+
+        actions = [(C, C), (C, C), (C, C), (C, C), (D, C)]
+        self.versus_test(axelrod.Cooperator(), expected_actions=actions)
+
+        actions = [(C, D), (D, D), (D, D), (D, D), (D, D)]
+        self.versus_test(axelrod.Defector(), expected_actions=actions)
+
+        actions = [(C, C), (C, D), (D, C), (C, D), (D, C)]
```

Comment:
Could you add one more round to this `actions`, the point of having `match_attributes={"length":-1}` is to show how the strategy acts when playing without knowledge of the when the last round is so the last actions would be:

```python
actions = [(C, C), (C, D), (D, C), (C, D), (D, C), (C, D)]
```

Then can you repeat the exact same test but without `match_attributes` which should then have:

```python
actions = [(C, C), (C, D), (D, C), (C, D), (D, C), (D D)]
```

Which shows that the "last round" instruction is working. (So just move your other alternator test here and add the extra round).